server/src/main/java/org/apache/druid/server/http/DataSourcesResource.java

```diff
@@ -410,7 +414,74 @@ public Response getServedSegmentsInInterval(
       }
       return Response.ok(segmentIds).build();
     }
-    return getServedSegmentsInInterval(dataSourceName, full != null, theInterval::contains);
+    return getServedSegmentsInInterval(dataSource, full != null, theInterval::contains);
+  }
+
+  @GET
+  @Path("/{dataSourceName}/unused/intervals")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ResourceFilters(DatasourceResourceFilter.class)
+  public Response getIntervalsWithUnusedSegmentsOrAllUnusedSegmentsPerIntervals(
+      @PathParam("dataSourceName") String dataSourceName,
+      @QueryParam("simple") String simple,
+      @QueryParam("full") String full,
+      @QueryParam("limit") @Nullable Integer limit,
+      @QueryParam("offset") @Nullable Integer offset
+  )
+  {
+    Preconditions.checkArgument(limit == null || limit > 0, "limit must be > 0");
```

Comment:
Would suggest throwing `InvalidInput` DruidException instead of these pre-conditions

Reply:
good suggestion, fixed